service/worker/replicator/replicator.go

```diff
@@ -172,3 +182,65 @@ func (r *Replicator) listenToClusterMetadataChange() {
 		},
 	)
 }
+
+func (r *Replicator) cleanupAckedMessages(
+	ctx context.Context,
+	deletedMessageID int64,
+) (int64, error) {
+	ackLevelByCluster, err := r.namespaceReplicationQueue.GetAckLevels(ctx)
+	if err != nil {
+		return deletedMessageID, err
+	}
+
+	connectedClusters := r.clusterMetadata.GetAllClusterInfo()
+	highWatermark := deletedMessageID
+	connectedClustersLowWatermark := int64(math.MaxInt64)
+	for clusterName, ackLevel := range ackLevelByCluster {
+		if clusterName == r.clusterMetadata.GetCurrentClusterName() {
+			continue
+		}
+		if ackLevel > highWatermark {
+			highWatermark = ackLevel
+		}
+		if _, ok := connectedClusters[clusterName]; !ok {
+			continue
+		}
+		if ackLevel < connectedClustersLowWatermark {
```

Comment:
Is ackLevel inclusive or exclusive? If inclusive, we need to + 1 when deleting messages?

Reply:
historical reason this is exclusive